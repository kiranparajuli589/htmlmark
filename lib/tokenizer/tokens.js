const TOKENS = {
  newLine: "new-line",
  paragraph: "paragraph",
  codeBlock: "code-block",
  comment: "comment",
  image: "image",
  quote: "quote",
  bold: "bold",
  italic: "italic",
  underline: "underline",
  strikeThrough: "strike-through",
  code: "code",
  link: "link",
  listItem: "list-item",
  countItem: "count-item",
  heading: "heading",
  checkBox: "check-box",
  checkBoxChecked: "check-box-checked",
  text: "text"
}

module.exports = TOKENS
