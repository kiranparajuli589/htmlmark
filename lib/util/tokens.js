const TOKENS = {
  NEW_LINE: "new-line",
  PARAGRAPH: "paragraph",
  CODE_BLOCK: "code-block",
  COMMENT: "comment",
  IMAGE: "image",
  QUOTE: "quote",
  BOLD: "bold",
  ITALIC: "italic",
  UNDERLINE: "underline",
  STRIKE_THROUGH: "strike-through",
  CODE: "code",
  LINK: "link",
  LIST_ITEM: "list-item",
  COUNT_ITEM: "count-item",
  HEADING: "heading",
  CHECKBOX: "check-box",
  CHECKBOX_CHECKED: "check-box-checked",
  TEXT: "text",
  HR_LINE: "hr-line",
  TABLE: "table",
}

module.exports = TOKENS
