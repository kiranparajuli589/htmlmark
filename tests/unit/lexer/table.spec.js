import { Lexer } from "../../../lib/lexer/index.js"
import { TOKENS } from "../../../lib/util/tokens.js"


describe("table", () => {
	describe("is not a table", () => {
		it("only header", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |"
			])
			const tokens = lexer.run()
			expect(tokens).toMatchSnapshot()
		})
		it("only header and separator", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				"|---|---|"
			])
			const tokens = lexer.run()
			expect(tokens).toMatchSnapshot()
		})
		it("false separator", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				"| --- |--|",
				"| row 1 c1 | row 1 c2 |"
			])
			const tokens = lexer.run()
			expect(tokens).toMatchSnapshot()
		})
		it("not consistent cell count", () => {
			const lines = [
				"| column 1 | column 2 |",
				"| --- |",
				"| row 1 c1 | row 1 c2 |"
			]
			const lexer = new Lexer(lines)
			const tokens = lexer.run()
			expect(tokens.length).toBe(1)
			expect(tokens[0].type).toBe(TOKENS.PARAGRAPH)
		})
		it("other tokens in between 1", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				"",
				"|---|---|",
				"| row 1 c1 | row 1 c2 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})
		it("other tokens in between 2", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				"|---|---|",
				"",
				"| row 1 c1 | row 1 c2 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})
		it("other tokens in between 3", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				"|---|---|",
				"| row 1 c1 | row 1 c2 |",
				"",
				"| row 2 c1 | row 2 c2 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})
	})
	describe("indentation", () => {
		it("different indent header -> separator", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				"    |---|---|",
				"| row 1 c1 | row 1 c2 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})

		it("different indent header, separator -> body row", () => {
			const lexer = new Lexer([
				"    | column 1 | column 2 |",
				"    |---|---|",
				"| row 1 c1 | row 1 c2 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})

		it("the indented table", () => {
			const lexer = new Lexer([
				"  | column 1 | column 2 |",
				"  |---|---|",
				"  | row 1 c1 | row 1 c2 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})

		it("indent should break the table 1", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				"|---|---|",
				"| row 1 c1 | row 1 c2 |",
				"    | row 2 c1 | row 2 c2 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})
		it("indent should break the table 2", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				"|---|---|",
				"| row 1 c1 | row 1 c2 |",
				"    | column 1 | column 2 |",
				"    |---|---|",
				"    | row 1 c1 | row 1 c2 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})
	})
	describe("cell count", () => {
		it("cell count should break the table 1", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				"|---|---|",
				"| row 1 c1 | row 1 c2 |",
				"| row 2 c1 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})
		it("cell count should break the table 2", () => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				"|---|---|",
				"| row 1 c1 | row 1 c2 |",
				"| column 1 |",
				"|---|",
				"| row 1 c1 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})
	})
	describe("heading body separator", () => {
		it.each([
			"|-------|-------|",
			"|:-------:|:---------:|",
			"| :-------: | :---------: |",
			"| ------- | --------- |"
		])("should parse a table with separator: '%s'", (line) => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				line,
				"| row 1 c1 | row 1 c2 |",
				"| row 2 c1 | row 2 c2 |"
			])
			expect(lexer.run()).toMatchSnapshot()
		})
		it.each([
			"|-|-|",
			"|:|:|",
			"| :|: |",
			"|-| |",
			"|-|:|",
			"|-|: |",
			"|:|-|",
			"| :|-|"
		])("should not parse as table with invalid separator: '%s'", (line) => {
			const lexer = new Lexer([
				"| column 1 | column 2 |",
				line,
				"| row 1 c1 | row 1 c2 |"
			])
			const lexed = lexer.run()
			expect(lexed.length).toBe(1)
			expect(lexed.at(0).type).toBe(TOKENS.PARAGRAPH)
		})
	})
})
