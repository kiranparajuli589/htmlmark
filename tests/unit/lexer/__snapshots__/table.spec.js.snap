// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table cell count cell count should break the table 1 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
  Object {
    "indent": 0,
    "raw": "| row 2 c1 |",
    "tokens": Array [
      Object {
        "raw": "| row 2 c1 |",
        "type": "text",
        "value": "| row 2 c1 |",
      },
    ],
    "type": "paragraph",
    "value": "| row 2 c1 |",
  },
]
`;

exports[`table cell count cell count should break the table 2 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
]
`;

exports[`table heading body separator should not parse as table with invalid separator: '|-|-|' 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": "-",
          "tokens": Array [
            Object {
              "raw": "-",
              "type": "text",
              "value": "-",
            },
          ],
        },
        Object {
          "raw": "-",
          "tokens": Array [
            Object {
              "raw": "-",
              "type": "text",
              "value": "-",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": false,
  },
]
`;

exports[`table heading body separator should parse a table with separator: '| :-------: | :---------: |' 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 2 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 2 c1",
              "type": "text",
              "value": "row 2 c1",
            },
          ],
        },
        Object {
          "raw": " row 2 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 2 c2",
              "type": "text",
              "value": "row 2 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
]
`;

exports[`table heading body separator should parse a table with separator: '| ------- | --------- |' 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 2 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 2 c1",
              "type": "text",
              "value": "row 2 c1",
            },
          ],
        },
        Object {
          "raw": " row 2 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 2 c2",
              "type": "text",
              "value": "row 2 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
]
`;

exports[`table heading body separator should parse a table with separator: '|:-------:|:---------:|' 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 2 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 2 c1",
              "type": "text",
              "value": "row 2 c1",
            },
          ],
        },
        Object {
          "raw": " row 2 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 2 c2",
              "type": "text",
              "value": "row 2 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
]
`;

exports[`table heading body separator should parse a table with separator: '|-------|-------|' 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 2 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 2 c1",
              "type": "text",
              "value": "row 2 c1",
            },
          ],
        },
        Object {
          "raw": " row 2 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 2 c2",
              "type": "text",
              "value": "row 2 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
]
`;

exports[`table indentation different indent header -> separator 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "| column 1 | column 2 |
    |---|---|
| row 1 c1 | row 1 c2 |",
    "tokens": Array [
      Object {
        "raw": "| column 1 | column 2 |     |---|---| | row 1 c1 | row 1 c2 |",
        "type": "text",
        "value": "| column 1 | column 2 |     |---|---| | row 1 c1 | row 1 c2 |",
      },
    ],
    "type": "paragraph",
    "value": "| column 1 | column 2 |     |---|---| | row 1 c1 | row 1 c2 |",
  },
]
`;

exports[`table indentation different indent header, separator -> body row 1`] = `
Array [
  Object {
    "indent": 4,
    "language": null,
    "raw": "    | column 1 | column 2 |
    |---|---|",
    "type": "code-block",
    "value": "| column 1 | column 2 |
|---|---|",
  },
  Object {
    "indent": 0,
    "raw": "| row 1 c1 | row 1 c2 |",
    "tokens": Array [
      Object {
        "raw": "| row 1 c1 | row 1 c2 |",
        "type": "text",
        "value": "| row 1 c1 | row 1 c2 |",
      },
    ],
    "type": "paragraph",
    "value": "| row 1 c1 | row 1 c2 |",
  },
]
`;

exports[`table indentation indent should break the table 1 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
  Object {
    "indent": 4,
    "raw": "    | row 2 c1 | row 2 c2 |",
    "tokens": Array [
      Object {
        "raw": "    | row 2 c1 | row 2 c2 |",
        "type": "text",
        "value": "    | row 2 c1 | row 2 c2 |",
      },
    ],
    "type": "paragraph",
    "value": "    | row 2 c1 | row 2 c2 |",
  },
]
`;

exports[`table indentation indent should break the table 2 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
  Object {
    "indent": 4,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
]
`;

exports[`table indentation the indented table 1`] = `
Array [
  Object {
    "indent": 2,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
]
`;

exports[`table is not a table false separator 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " --- ",
          "tokens": Array [
            Object {
              "raw": "---",
              "type": "text",
              "value": "---",
            },
          ],
        },
        Object {
          "raw": "--",
          "tokens": Array [
            Object {
              "raw": "--",
              "type": "text",
              "value": "--",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": false,
  },
]
`;

exports[`table is not a table only header 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "| column 1 | column 2 |",
    "tokens": Array [
      Object {
        "raw": "| column 1 | column 2 |",
        "type": "text",
        "value": "| column 1 | column 2 |",
      },
    ],
    "type": "paragraph",
    "value": "| column 1 | column 2 |",
  },
]
`;

exports[`table is not a table only header and separator 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "| column 1 | column 2 |
|---|---|",
    "tokens": Array [
      Object {
        "raw": "| column 1 | column 2 | |---|---|",
        "type": "text",
        "value": "| column 1 | column 2 | |---|---|",
      },
    ],
    "type": "paragraph",
    "value": "| column 1 | column 2 | |---|---|",
  },
]
`;

exports[`table is not a table other tokens in between 1 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "| column 1 | column 2 |",
    "tokens": Array [
      Object {
        "raw": "| column 1 | column 2 |",
        "type": "text",
        "value": "| column 1 | column 2 |",
      },
    ],
    "type": "paragraph",
    "value": "| column 1 | column 2 |",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "indent": 0,
    "raw": "|---|---|
| row 1 c1 | row 1 c2 |",
    "tokens": Array [
      Object {
        "raw": "|---|---| | row 1 c1 | row 1 c2 |",
        "type": "text",
        "value": "|---|---| | row 1 c1 | row 1 c2 |",
      },
    ],
    "type": "paragraph",
    "value": "|---|---| | row 1 c1 | row 1 c2 |",
  },
]
`;

exports[`table is not a table other tokens in between 2 1`] = `
Array [
  Object {
    "indent": 0,
    "raw": "| column 1 | column 2 |
|---|---|",
    "tokens": Array [
      Object {
        "raw": "| column 1 | column 2 | |---|---|",
        "type": "text",
        "value": "| column 1 | column 2 | |---|---|",
      },
    ],
    "type": "paragraph",
    "value": "| column 1 | column 2 | |---|---|",
  },
  Object {
    "type": "new-line",
  },
  Object {
    "indent": 0,
    "raw": "| row 1 c1 | row 1 c2 |",
    "tokens": Array [
      Object {
        "raw": "| row 1 c1 | row 1 c2 |",
        "type": "text",
        "value": "| row 1 c1 | row 1 c2 |",
      },
    ],
    "type": "paragraph",
    "value": "| row 1 c1 | row 1 c2 |",
  },
]
`;

exports[`table is not a table other tokens in between 3 1`] = `
Array [
  Object {
    "indent": 0,
    "rows": Array [
      Array [
        Object {
          "raw": " column 1 ",
          "tokens": Array [
            Object {
              "raw": "column 1",
              "type": "text",
              "value": "column 1",
            },
          ],
        },
        Object {
          "raw": " column 2 ",
          "tokens": Array [
            Object {
              "raw": "column 2",
              "type": "text",
              "value": "column 2",
            },
          ],
        },
      ],
      Array [
        Object {
          "raw": " row 1 c1 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c1",
              "type": "text",
              "value": "row 1 c1",
            },
          ],
        },
        Object {
          "raw": " row 1 c2 ",
          "tokens": Array [
            Object {
              "raw": "row 1 c2",
              "type": "text",
              "value": "row 1 c2",
            },
          ],
        },
      ],
    ],
    "type": "table",
    "withHeading": true,
  },
  Object {
    "type": "new-line",
  },
  Object {
    "indent": 0,
    "raw": "| row 2 c1 | row 2 c2 |",
    "tokens": Array [
      Object {
        "raw": "| row 2 c1 | row 2 c2 |",
        "type": "text",
        "value": "| row 2 c1 | row 2 c2 |",
      },
    ],
    "type": "paragraph",
    "value": "| row 2 c1 | row 2 c2 |",
  },
]
`;
